### PR TITLE
Steem Tweaks for Shadow Build and Bug Fixes

### PR DESCRIPTION
**Pull request by swainn**
Primary changes in this Pull Request:

- Namespace loggers with "tethys" so they will log to Tethys log handler (console)
- Adds generalized `FileCollectionsControllerMixin` from STEEM - methods for controllers to help manage Resources with FileCollections relationships.
- Make it so basemap dropdown on `MapView` doesn't get too long if there are many basemaps enabled.
- Add `app` as an addition argument to `ResourceCondorWorkflow`
- Fix overflow issues with long nav header titles
- Fix bug with `ResourceSpatialManager.delete_extent_layer()`
- Move extent-layer-related and other general methods from `SteemSpatialManager` to `ResourceSpatialManager`.
- Update tests accordingly

Please review the checklist before submitting the Pull Request:

- [x] Pull request has a meaning full name (not just the commit message from of your last commit)
- [x] Code has been linted using flake8
- [x] All methods have accurate Google-Style Docstrings
- [x] 100% test coverage for new content
- [x] Tests for each common use case (please don't write one test that covers all use cases)
- [ ] Bonus: Use TypeHints

### CODE REVIEW

File: tethysext/atcore/services/resource_spatial_manager.py [hunk 1]
```diff
@@ -59,6 +76,44 @@ def get_extent_for_project(self, datastore_name, resource_id):
...
+        """
+        Get url for map preview image.
+        Returns:
+            str: preview image url.
+        """
+        # Default image url
+        layer_preview_url = None
+        layer_name = f'{self.WORKSPACE}:{self.get_extent_layer_name(resource.id)}'
+        try:
+            extent = self.get_extent_for_project(datastore_name=f'{self.WORKSPACE}_primary_db',
```
Comment (msouff) [inline tethysext/atcore/services/resource_spatial_manager.py:89]: is there an easy way to get the datastore_name from the app.DATABASE variable in app.py? If I remember well they will always be the same.

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 2]
```diff
@@ -18,12 +18,13 @@ class ResourceCondorWorkflow(object):
     """
     Helper class that prepares and submits the new project upload jobs and workflow.
     """
-    def __init__(self, user, workflow_name, workspace_path, resource_db_url, resource,
+    def __init__(self, app, user, workflow_name, workspace_path, resource_db_url, resource,
```
Comment (msouff) [inline tethysext/atcore/services/resource_condor_workflow.py:21]: Why are we adding the app as a parameter?

File: tethysext/atcore/mixins/file_collection_controller_mixin.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            for unit in ['Bytes', 'KB', 'MB', 'GB', 'TB']:
+                if total_size < 1024:
+                    break
+                total_size /= 1024.0
+            total_size = f"{total_size:.2f} {unit}"
+
+            # Create file collection details tuple
+            file_collection_details = ('File Collection Details', {
+                'ID': file_collection.id,
+                'File count': file_count,
```
[the remark this answers is not in the document]
Comment (swainn) [reply tethysext/atcore/mixins/file_collection_controller_mixin.py:70]: Done.

File: tethysext/atcore/services/resource_spatial_manager.py (hunk 1 above)
Comment (swainn) [reply tethysext/atcore/services/resource_spatial_manager.py:89]: They do not need to be the same. The datastore name is not the same as the connection information (name of the database). It is an arbitrary name that happens to be the same in our apps.

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 2 above)
Comment (swainn) [reply tethysext/atcore/services/resource_condor_workflow.py:21]: It's needed by the downstream Workflow classes in STEEM and Shadow. You'll see in the steem_adapter PR.

File: tethysext/atcore/services/resource_spatial_manager.py (hunk 1 above)
Comment (swainn) [reply tethysext/atcore/services/resource_spatial_manager.py]: Changed this to use the DATASTORE property that is now part of the class.

**Pull request merged**